### ISSUE TITLE
Common issues with installing BindCraft - please add to Readme

### ISSUE THREAD

**Issue by linuxfold** [martinpacesa/BindCraft#123]
I was unable to install with the included install script. It seemed to try to solve forever. I am using CUDA 12.7.

I tried running the test command `python -u ./bindcraft.py --settings './settings_target/PDL1.json' --filters './settings_filters/default_filters.json' --advanced './settings_advanced/default_4stage_multimer.json'` and installing everything that said was missing. 

This is the output after installing everything I could and then running the test command:
Available GPUs:
NVIDIA GeForce RTX 40901: gpu
[....]
Traceback (most recent call last):
  File "/home/user/Documents/BindCraft/./bindcraft.py", line 41, in <module>
    design_paths = generate_directories(target_settings["design_path"])
  File "/home/user/Documents/BindCraft/functions/generic_utils.py", line 50, in generate_directories
    os.makedirs(path, exist_ok=True)
  File "/home/user/anaconda3/envs/BindCraft/lib/python3.10/os.py", line 215, in makedirs
    makedirs(head, exist_ok=exist_ok)
  File "/home/user/anaconda3/envs/BindCraft/lib/python3.10/os.py", line 215, in makedirs
    makedirs(head, exist_ok=exist_ok)
  File "/home/user/anaconda3/envs/BindCraft/lib/python3.10/os.py", line 215, in makedirs
    makedirs(head, exist_ok=exist_ok)
  [Previous line repeated 2 more times]
  File "/home/user/anaconda3/envs/BindCraft/lib/python3.10/os.py", line 225, in makedirs
    mkdir(name, mode)
PermissionError: [Errno 13] Permission denied: '/content'

**Comment by linuxfold** [comment]
I think I figured out that issue... having to edit the json file to point to the correct paths.

Now when I run I get the following output after the GPUs:

┌──────────────────────────────────────────────────────────────────────────────┐
│                                 PyRosetta-4                                  │
│              Created in JHU by Sergey Lyskov and PyRosetta Team              │
│              (C) Copyright Rosetta Commons Member Institutions               │
│                                                                              │
│ NOTE: USE OF PyRosetta FOR COMMERCIAL PURPOSES REQUIRE PURCHASE OF A LICENSE │
│         See LICENSE.PyRosetta.md or email license@uw.edu for details         │
└──────────────────────────────────────────────────────────────────────────────┘
PyRosetta-4 2024 [Rosetta PyRosetta4.Release.python310.ubuntu 2024.42+release.3366cf78a3df04339d1982e94531b77b098ddb99 2024-10-11T08:24:04] retrieved from: http://www.pyrosetta.org
Starting trajectory: PDL1_l140_s329539
WARNING: 'model_1_multimer_v3' not found
WARNING: 'model_2_multimer_v3' not found
WARNING: 'model_3_multimer_v3' not found
WARNING: 'model_4_multimer_v3' not found
WARNING: 'model_5_multimer_v3' not found
Stage 1: Test Logits
Traceback (most recent call last):
  File "/home/user/Documents/BindCraft/./bindcraft.py", line 106, in <module>
    trajectory = binder_hallucination(design_name, target_settings["starting_pdb"], target_settings["chains"],
  File "/home/user/Documents/BindCraft/functions/colabdesign_utils.py", line 97, in binder_hallucination
    af_model.design_logits(iters=50, e_soft=0.9, models=design_models, num_models=1, sample_models=advanced_settings["sample_models"], save_best=True)
  File "/home/user/anaconda3/envs/BindCraft/lib/python3.10/site-packages/colabdesign/af/design.py", line 346, in design_logits
    self.design(iters, **kwargs)
  File "/home/user/anaconda3/envs/BindCraft/lib/python3.10/site-packages/colabdesign/af/design.py", line 340, in design
    self.step(lr_scale=lr_scale, num_recycles=num_recycles,
  File "/home/user/anaconda3/envs/BindCraft/lib/python3.10/site-packages/colabdesign/af/design.py", line 206, in step
    self.run(num_recycles=num_recycles, num_models=num_models, sample_models=sample_models,
  File "/home/user/anaconda3/envs/BindCraft/lib/python3.10/site-packages/colabdesign/af/design.py", line 89, in run
    model_nums = self._get_model_nums(num_models, sample_models, models)
  File "/home/user/anaconda3/envs/BindCraft/lib/python3.10/site-packages/colabdesign/af/design.py", line 71, in _get_model_nums
    ns = [ns[n if isinstance(n,int) else ns_name.index(n)] for n in models]
  File "/home/user/anaconda3/envs/BindCraft/lib/python3.10/site-packages/colabdesign/af/design.py", line 71, in <listcomp>
    ns = [ns[n if isinstance(n,int) else ns_name.index(n)] for n in models]
IndexError: list index out of range

**Comment by linuxfold** [comment]
Was able to solve this problem with issue #76 .

Had to edit the af_params_dir in default_4stage_multimer.json to point to the params.

I think it is working now. Got this error but it seems to be proceeding:

```
2024-12-08 18:17:33.591370: W external/xla/xla/service/gpu/ir_emitter_unnested.cc:1171] Unable to parse backend config for custom call: Could not convert JSON string to proto: Expected : between key:value pair.
= true, full_matrice
^
Fall back to parse the raw backend config str.
```
It outputted the following:
Trajectory successful, final pLDDT: 0.93
Starting trajectory took: 0 hours, 5 minutes, 7 seconds

Then I got this error:

```
Traceback (most recent call last):
  File "/home/user/Documents/BindCraft/./bindcraft.py", line 169, in <module>
    mpnn_trajectories = mpnn_gen_sequence(trajectory_pdb, binder_chain, trajectory_interface_residues, advanced_settings)
  File "/home/user/Documents/BindCraft/functions/colabdesign_utils.py", line 345, in mpnn_gen_sequence
    mpnn_model = mk_mpnn_model(backbone_noise=advanced_settings["backbone_noise"], model_name=advanced_settings["model_path"], weights=advanced_settings["mpnn_weights"])
TypeError: mk_mpnn_model.__init__() got an unexpected keyword argument 'weights'
```

**Comment by linuxfold** [comment]
A few hours later... and found from #28 

 pip uninstall colabdesign

 pip install git+https://github.com/sokrypton/ColabDesign

fixes TypeError: mk_mpnn_model.__init__() got an unexpected keyword argument 'weights'

**Comment by linuxfold** [comment]
Adding another error:

ERROR: Flag 'coordination_agent_recoverable' was defined more than once but with differing types. Defined in files 'external/xla/xla/tsl/distributed_runtime/coordination/coordination_service_agent.cc' and 'external/xla/xla/tsl/distributed_runtime/coordination/coordination_service_agent.cc'.

Fixed by: 

# inside the BindCraft environment
```
python -m pip uninstall -y \
  jaxlib jax jax-cuda12-plugin jax-cuda12-pjrt

# optional but helps:
pip cache purge
```

then

```   
pip install --upgrade "jax[cuda12_local]"  
```

**Comment by traversaro** [comment]
Just for reference, the error:

> ERROR: Flag 'coordination_agent_recoverable' was defined more than once but with differing types. Defined in files 'external/xla/xla/tsl/distributed_runtime/coordination/coordination_service_agent.cc' and 'external/xla/xla/tsl/distributed_runtime/coordination/coordination_service_agent.cc'.

was a bug in the jaxlib conda-forge package (see https://github.com/conda-forge/jaxlib-feedstock/issues/320) that is now fixed, so the latest build of `jaxlib=0.7.0=*_201` should work fine.
